armi/tests/test_plugins.py

```diff
@@ -42,13 +43,25 @@ def defineFlags():
         return {"SUPER_FLAG": utils.flags.auto()}
 
 
+class SillyAxialExpansionChanger(AxialExpansionChanger):
+    pass
+
+
+class SillyAxialPlugin(plugins.ArmiPlugin):
+    @staticmethod
+    @plugins.HOOKIMPL
+    def getAxialExpansionChanger() -> type[SillyAxialExpansionChanger]:
+        return SillyAxialExpansionChanger
+
+
 class TestPluginRegistration(unittest.TestCase):
     def setUp(self):
         """
         Manipulate the standard App. We can't just configure our own, since the
         pytest environment bleeds between tests.
         """
-        self._backupApp = deepcopy(getApp())
+        self.app = getApp()
+        self._backupApp = deepcopy(self.app)
```

Comment:
what's `self._backupApp` for?

Reply:
Yeah this was a weird one. `getApp` returns the global armi application. The intent of these tests is to modify the app so we can register plugins and look for behavior changes. But that global app, and any changes to it, would carry over from test to test. So these tests attempt to restore the global app state when they are done

Reply:
Tagging in @john-science as this is getting out of my scope of practice

Reply:
This was just someone's attempt to find a way to test plugins.  The problem is that to TEST plugin tooling, you are going to want to register a bunch of plugins.  But if you do that, it will stomp all over every unit test, because they all share a default `App()`.

So this was an attempt to have the plugin infrastructure tests be isolated from stomping all over every other unit tests in ARMI.

You should absolutely not make the change you are making here. Leave it as it is, please.

Reply:
Okay, I think your change doesn't break anything.

But it certainly makes understanding the already complicated test logic harder.  I don't love that.

Reply:
@john-science I'm going to put the resolution of this in your court, if you don't mind!

Reply:
> Okay, I think your change doesn't break anything.
> 
> But it certainly makes understanding the already complicated test logic harder. I don't love that.

Happy to revert. Tried to remove the number of `deepcopy` and `getApp` calls. But I agree mucking around with these global app states is not ideal. There's probably a way we can test against a generic application without the global state. That should probably be a separate ticket though, but it would make these tests cleaner and more maintainable